### PR TITLE
Invoke should not run on removed repos

### PR DESCRIPTION
**Pull request by nhunzaker**
This commit fixes a bug where Microcosm would raise an exception if a
fork was removed from history as the result of a parent changing. It
also adds an assertion to make it clearer during testing.

Sorry, @dce :(

### CODE REVIEW

File: test/unit/history/invoke.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    jest.spyOn(child, 'release')
+
+    parent.addDomain('test', {
+      getInitialState() {
+        return false
+      }
+    })
+
+    expect(child.release).not.toHaveBeenCalled()
+  })
```
Comment (mackermedia) [inline test/unit/history/invoke.test.js]: I'm not super familiar with the lifecycle hooks, but reading this test it isn't obvious to me why the child goes away (child.teardown()?) triggered by adding a domain on the parent? And if so, why would `child.release` not be called? What is `child.release` ?

**Comment by mackermedia** [comment]
clarifying question, but 👍

File: test/unit/history/invoke.test.js (hunk 1 above)
Comment (nhunzaker) [inline test/unit/history/invoke.test.js]: Still waiting for some details on that, however I was investigating and found this edge case, so I went ahead and stamped it out.

So we'll see :)

Comment (nhunzaker) [inline test/unit/history/invoke.test.js]: As for `release`, `release` is an internal Microcosm method for sending out a change event. It fires a change, then it invokes Effects with the related action.

This allows a tree of microcosms (parent + forks) to all resolve an action, then handle change subscriptions when every microcosm is up to date.
